IOMC/EventVertexGenerators/src/BaseEvtVtxGenerator.cc

```diff
@@ -47,20 +49,39 @@ void BaseEvtVtxGenerator::produce(Event& evt, const EventSetup&) {
 
   Handle<HepMCProduct> HepUnsmearedMCEvt;
 
-  evt.getByToken(sourceToken, HepUnsmearedMCEvt);
+  bool found = evt.getByToken(sourceToken, HepUnsmearedMCEvt);
+
+  if (found) {  // HepMC event exists
+
+    // Make a copy
+    HepMC::GenEvent* genevt = new HepMC::GenEvent(*HepUnsmearedMCEvt->GetEvent());
```

Comment:
@mkirsano , is it really needed making genevt via "new", I would try to create  edm::HepMCProduct directly.